### PR TITLE
build: Streamline `mocha` config

### PR DESCRIPTION
**Pull request by danielbankhead**
Adds support for native `mocha` config support - now we can use `npx mocha` without passing additional args/config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,13 +26,16 @@
     "google storage",
     "storage"
   ],
+  "mocha": {
```
Comment (ddelgrosso1) [inline package.json:29]: What does moving this here vs the command line provide?

Comment (ddelgrosso1) [inline package.json:29]: What does moving this here vs the command line provide?

Comment (danielbankhead) [reply package.json:29]: It allows 2 things:
- `npx mocha <path-to-test(s)>` without additional flags to type/remember
- Easier IDE integration - IDEs no long require additional config in order to run `mocha` tests (which can change over time)

**Comment by danielbankhead** [comment]
Additional note/context: Mocha merges the config file (`.mocharc` - owned by owl-bot/synthtool) and `package.json#mocha` (owned by us): https://mochajs.org/#merging

File: package.json (hunk 1 above)
Comment (danielbankhead) [reply package.json:29]: Add, if `mocha` is globally installed (`npm i -g mocha`) simply typing `mocha` in the CLI will run the default tests (same as `npx mocha`)

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**